src/services/1inch.js

```diff
@@ -59,11 +58,29 @@ const ethProvider = () => {
   return getEthereumProvider(getEnv().NETWORK_PROVIDER);
 };
 
+export const get1InchSpenderAddress = async (): Promise<?string> => {
+  const response = await axios.get(`${EXCHANGE_URL}/approve/spender`).catch((error) => error);
+
+  const spenderAddress = response?.data?.address;
+
+  if (!spenderAddress) {
+    reportLog('get1inchApproveAddress API call failed', { response });
+    return null;
+  }
+
+  return spenderAddress;
+};
+
 const getAllowanceSet = async (clientAddress: string, safeFromAddress: string, fromAsset: Asset) => {
   let allowanceSet = true;
   if (fromAsset.code !== ETH) {
     const assetContract = new ethers.Contract(safeFromAddress, ERC20_CONTRACT_ABI, ethProvider());
-    const allowance: BigNumber = await assetContract.allowance(clientAddress, EXCHANGE_ADDRESS);
+    const exchangeAddress = await get1InchSpenderAddress();
+    if (!exchangeAddress) {
+      reportLog('getAllowanceSet -> get1inchApproveAddress failed');
```

Comment:
Could we report the error also coming back from `/approve/sender`? From what i can see being return in the `.catch` block from `get1InchSpenderAddress`, you can just report `exchangeAddress`?

Reply:
The error is reported as part of response on `get1InchSpenderAddress`.